src/risedevtool/src/task/compactor_service.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+use anyhow::{anyhow, Result};
+
+use crate::util::{get_program_args, get_program_env_cmd, get_program_name};
+use crate::{CompactorConfig, ExecuteContext, Task};
+
+pub struct CompactorService {
+    config: CompactorConfig,
+}
+
+impl CompactorService {
+    pub fn new(config: CompactorConfig) -> Result<Self> {
+        Ok(Self { config })
+    }
+
+    fn compactor(&self) -> Result<Command> {
+        let prefix_bin = env::var("PREFIX_BIN")?;
+
+        if let Ok(x) = env::var("ENABLE_ALL_IN_ONE") && x == "true" {
+            Ok(Command::new(Path::new(&prefix_bin).join("risingwave").join("compactor")))
+        } else {
+            Ok(Command::new(Path::new(&prefix_bin).join("compactor")))
+        }
+    }
+
+    /// Apply command args accroding to config
+    pub fn apply_command_args(cmd: &mut Command, config: &CompactorConfig) -> Result<()> {
+        cmd.arg("--host")
+            .arg(format!("{}:{}", config.address, config.port))
+            .arg("--prometheus-listener-addr")
+            .arg(format!(
+                "{}:{}",
+                config.exporter_address, config.exporter_port
+            ))
+            .arg("--metrics-level")
+            .arg("1");
+
+        let provide_minio = config.provide_minio.as_ref().unwrap();
+        let provide_aws_s3 = config.provide_aws_s3.as_ref().unwrap();
+        match (provide_minio.as_slice(), provide_aws_s3.as_slice()) {
+            ([], []) => {
+                return Err(anyhow!(
+                    "Compactor is not compatible with in_memory_kv_state_backend. Need to enable either minio or aws-s3.",
```

Comment:
`in_memory_kv_state_backend` -> in-memory state backend. There are two types of in-memory state backend: the BTreeMap KV based one, and the hummock + in-memory one.